src/mesh/distributed_mesh.C

```diff
@@ -213,6 +229,14 @@ DistributedMesh::DistributedMesh (const UnstructuredMesh & other_mesh) :
   this->update_parallel_id_counts();
 }
 
+void DistributedMesh::move_nodes_and_elements(MeshBase && other_mesh)
+{
+  this->_nodes = std::move((libmesh_cast_ptr<DistributedMesh*>(&other_mesh))->_nodes);
+  this->_n_nodes = dof_id_type(_nodes.size());
```

Comment:
No, it is not.  Likewise for this->_n_elem.  On a distributed mesh we aren't storing every single DofObject.  Just copy the other mesh's caches.  We wouldn't have them if they were redundant.

Reply:
This PR has way too many embarrassing oversights. Should have looked for the cached methods.